### ISSUE TITLE
is-docker-installed?

### ISSUE THREAD

**Issue by fwhoami** [microsoft/vscode-docker#3711]
Does this occur consistently? Yes
Repro steps:

1. The extension docker is not working. Gives the error "Failed to connect. Is Docker installed?" when docker and docker-desktop are installed and working fine. The "Dev Containers" extension is also installed.
2.And it is also impossible to use the terminal in vs code.

Version: 1.22.2
OS: linux
OS Release: 5.15.76-1-MANJARO
Product: Visual Studio Code
Product Version: 1.70.0
Language: en

**Comment by karolz-ms** [comment]
@fwhoami can you elaborate on

>    2.And it is also impossible to use the terminal in vs code.

What errors are you seeing?

This makes me think something is wrong with the VS Code installation. Perhaps re-installation of VS Code will help (the latest version is 1.73.1, you have 1.70.0, a few months old).

**Comment by karolz-ms** [comment]
@fwhoami can you access Docker CLI from outside of VS Code (from standalone Linux terminal)? What is the path to `docker` executable? What does VS Code terminal "see" as `$PATH` and does it include path to `docker` executable?

You can tell the extension where `docker` executable is via `docker.dockerPath` setting, but this is last-resource workaround. It would be better if VS Code/VS Code terminal could discover `docker` on `$PATH` normally.

**Comment by fwhoami** [comment]
@karolz-ms yes,  standalone Linux terminal works fine with docker. How can you solve the problem with the help of docker.dockerPath setting, can you please tell me in more detail?

**Comment by karolz-ms** [comment]
Sure, here is the documentation on [how to edit VS Code settings](https://code.visualstudio.com/docs/getstarted/settings).

In there you can add a line that says
```json
"docker.dockerPath": "/path/to/your/docker/executable"
```

That should take care of the extension not being able to find the `docker` executable, but I suspect something is wrong with `$PATH` that VS Code is seeing, so it would be preferable to fix that and then the Docker extension would "just work" without this setting.

**Comment by fwhoami** [comment]
Unfortunately, this did not solve the problem.
>"docker.dockerPath": "/path/to/your/docker/executable"

**Comment by karolz-ms** [comment]
Can you elaborate?
